internal/distro/distro.go

```diff
@@ -118,9 +118,10 @@ type ImageType interface {
 
 	// Returns an osbuild manifest, containing the sources and pipeline necessary
 	// to build an image, given output format with all packages and customizations
-	// specified in the given blueprint. The packageSpecSets must be labelled in
-	// the same way as the originating PackageSets.
-	Manifest(b *blueprint.Customizations, options ImageOptions, repos []rpmmd.RepoConfig, packageSpecSets map[string][]rpmmd.PackageSpec, containers []container.Spec, seed int64) (Manifest, error)
+	// specified in the given blueprint; it also returns any warnings (e.g.
+	// deprecation notices) generated by the manifest.
+	// The packageSpecSets must be labelled in the same way as the originating PackageSets.
+	Manifest(b *blueprint.Customizations, options ImageOptions, repos []rpmmd.RepoConfig, packageSpecSets map[string][]rpmmd.PackageSpec, containers []container.Spec, seed int64) (Manifest, []string, error)
```

Comment:
I don't really like returning an extra value for warnings as a slice of strings though I admit that I don't see another straightforward way to do it right now.  Maybe we can have a result object here that encapsulates warning and error, but that's a bigger change that we might want to look at in a followup.

Not asking for a change here, just sharing thoughts for potential future changes.

Reply:
I agree. What could help is to use named return variables to make it obvious what's the purpose of the returned value. So something like:

```go
Manifest(b *blueprint.Customizations, options ImageOptions, repos []rpmmd.RepoConfig, packageSpecSets map[string][]rpmmd.PackageSpec, containers []container.Spec, seed int64) (m Manifest, warn []string, err error)
```